downgrade

```diff
@@ -448,12 +425,41 @@ parse_options() {
 }
 
 cli() {
-  parse_options "$@"
+  declare -a system_args
+
+  # get system and command-line arguments and parse everything
+  IFS=' ' read -r -a system_args <<<"$(read_downgrade_conf)"
+  parse_options "${system_args[@]}" "$@"
+
+  # make arrays unique
+  IFS=' ' read -r -a terms <<<"$(printf "%s\n" "${terms[@]}" | uniq | tr '\n' ' ')"
+  IFS=' ' read -r -a PACMAN_CACHE <<<"$(printf "%s\n" "${PACMAN_CACHE[@]}" | uniq | tr '\n' ' ')"
```

Comment:
```sh
read_unique() {
  local var=$1
  shift

  IFS=$'\n' read -r -a var <<<"$(printf "%s\n" "$@" | uniq)"
}
```

```suggestion
  read_unique terms "${terms[@]}"
  read_unique PACMAN_CACHE "${PACMAN_CACHE[@]}"
```

Reply:
Thanks :+1:

Reply:
Used `mapfile` instead because `read` was not able to parse newlines, not sure why. Also changed `uniq` to `sort -u` since `uniq` only considers adjacent dupicates and can therefore fail on several cases. This version works:

https://github.com/pbrisbin/downgrade/blob/e99d30aad608e9412e72ddffea2e792d92ef1b04/downgrade#L40-L46